### ISSUE TITLE
False positive unused import when import is referencing extension function and is used only in kdoc

### ISSUE THREAD

**Issue by matejdro** [pinterest/ktlint#1282]
## Steps to Reproduce
Run ktlint on the following two files (each file needs to be in a directory corresponding to the package):

```kotlin
package some.pkg

fun String.returnSelf(): String {
    return this
}

```

```kotlin
package other.pkg

import some.pkg.returnSelf

/**
 * Do not forget that you can also return string via [String.returnSelf]
 */
fun test() {}

```

## Expected Behavior

Ktlint should pass

## Observed Behavior

Ktlint fails with `FileB.kt:1:1 Unused import`

## Your Environment

* Version of ktlint used: `0.43.0`
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): `JLLeitschuh/ktlint-gradle 10.2.0`
* Version of Gradle used (if applicable): `7.2`
* Operating System and version: `5.15.1-1-MANJARO #1 SMP PREEMPT Sat Nov 6 14:11:10 UTC 2021 x86_64 GNU/Linux`

**Comment by paul-dingemans** [comment]
Can you provide some background of the use case for which you want to keep the import when it is only used for a reference in a KDoc? IntelliJ IDEA Ultimate has no problem to resolve the reference without the imports.

**Comment by matejdro** [comment]
> IntelliJ IDEA Ultimate has no problem to resolve the reference without the imports.

Hm, it does not resolve it for me without the import:

![image](https://user-images.githubusercontent.com/507922/141733925-b32e81c6-d307-4aca-b270-6d3b17a730ba.png)

**Comment by matejdro** [comment]
It happens with the latest one `212-1.6.0-release-799-IJ547.46`. 

Are you sure your IntelliJ can resolve this kdoc if you remove an import? Can you take a screenshot? Because I don't even see how would that work (IntelliJ needs imports to resolve conflicts etc.)

**Comment by paul-dingemans** [comment]
I think I do understand the miscommunication between us. 

Is it correct that your files are in different directories? If files are both in the same directory then IntelliJ indeed has no problems to display the link inside the kdoc as files belong to same package. However, if the files are in different directories (e.g. not in the same package) then IntelliJ needs the import for displaying the link inside kdoc correctly.

**Comment by matejdro** [comment]
Sorry, you are correct. I thought that files having the different package implied that they are in the different directories (I know kotlin allows different packages in the same directory, but this is generally not the practice).

Updated the bug report to make that clear.

**Comment by paul-dingemans** [comment]
See commit above. It should be fixed with that.

**Comment by matejdro** [comment]
Thanks!
